src/sentry/api/endpoints/group_integration_details.py

```diff
@@ -266,12 +266,25 @@ def post(self, request: Request, group, integration_id) -> Response:
             )
 
         installation = integration.get_installation(organization_id=organization_id)
-        try:
-            data = installation.create_issue(request.data)
-        except IntegrationFormError as exc:
-            return Response(exc.field_errors, status=400)
-        except IntegrationError as e:
-            return Response({"non_field_errors": [str(e)]}, status=400)
+
+        with ProjectManagementEvent(
+            action_type=ProjectManagementActionType.CREATE_EXTERNAL_ISSUE, integration=integration
+        ).capture() as lifecycle:
+            lifecycle.add_extras(
+                {
+                    "provider": integration.provider,
+                    "integration_id": integration.id,
+                }
+            )
+
+            try:
+                data = installation.create_issue(request.data)
+            except IntegrationFormError as exc:
+                lifecycle.record_halt(exc)
+                return Response(exc.field_errors, status=400)
+            except IntegrationError as e:
+                lifecycle.record_failure(e)
+                return Response({"non_field_errors": [str(e)]}, status=400)
```

Comment:
Q: What are examples of `IntegrationErrors`, like is that a failure on our side ?

Reply:
IntegrationErrors are any errors from a provider that our integrations themselves can't classify as a "field" error. 

We have some internal handling in the base `IntegrationInstallation` class that checks any `ApiInvalidRequestError` responses our integration methods raise: https://github.com/getsentry/sentry/blob/6209d6fbf55839bb7a2f93ef65decbf495a64974/src/sentry/integrations/base.py#L484

Which call into the implementing integration class's `error_fields_from_json` method: https://github.com/getsentry/sentry/blob/6209d6fbf55839bb7a2f93ef65decbf495a64974/src/sentry/integrations/jira/integration.py#L491